File: frontend/rastreador-web/src/types/geofence.ts
export interface LatLngDto {
  lat: number;
  lng: number;
}

export interface GeofenceDto {
  id: number;
  name: string;
  vehicleId: number | null;
  points: LatLngDto[];
  alertOnEnter: boolean;
  alertOnExit: boolean;
  createdAt: string;
}

export interface GeofenceCreateDto {
  name: string;
  vehicleId?: number | null;
  points: LatLngDto[];
  alertOnEnter: boolean;
  alertOnExit: boolean;
}

File: backend/Rastreador.Api/Models/Geofence.cs
using NetTopologySuite.Geometries;

namespace Rastreador.Api.Models;

public class Geofence
{
    public int Id { get; set; }
    public int CompanyId { get; set; }
    public Company? Company { get; set; }
    public string Name { get; set; } = string.Empty;

    /// <summary>Null = aplica a todos os veículos da empresa.</summary>
    public int? VehicleId { get; set; }
    public Vehicle? Vehicle { get; set; }

    public Polygon Area { get; set; } = null!;
    public bool AlertOnEnter { get; set; } = true;
    public bool AlertOnExit { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
